docs/internals/ARCHITECTURE.md

```diff
@@ -1,97 +1,92 @@
+
+The forest of LSM trees is implemented as an on-disk functional data structure. The data is
 organized as a tree of on-disk blocks (a block is 0.5 MiB in size). Blocks refer to other blocks by
-their on-disk address and checksum of their content. There's a root block, called the superblock,
-from which the rest of the state is reachable.
+their on-disk address and checksum of their content. From a root block, called the superblock,
+the rest of the state is reachable.
 
-When applying prepares to local state, replica doesn't overwrite any existing blocks in the data
-file, and instead writes new blocks only, maintaining the state of the superblock in memory. Once in
-a while, the superblock is atomically flushed to storage, forming a new checkpoint. When a replica
-crashes and restarts, it of course looses access to its previous in-memory superblock. Instead, it
-reads the previous checkpoint/superblock from storage, and reconstructs the lost state by replaying
-the suffix of the log of prepares after that checkpoint. Determinism guarantees that the replica
-ends up in the exact same state.
+When applying prepares to local state, replicsa don't overwrite any existing blocks in the data
+file and instead write new blocks only. This maintains the state of the superblock in memory. Once in
+a while, the superblock is atomically flushed to storage forming a new checkpoint. When a replica
+crashes and restarts it loses access to its previous in-memory superblock. It reads the previous 
+checkpoint/superblock from storage and reconstructs the lost state by replaying the suffix of the 
+log of prepares after that checkpoint. Determinism guarantees that the replica ends up in the exact 
+same state.
 
 TigerBeetle assumes that replica's storage can fail. If a replica writes a prepare to the WAL or a
-block of an LSM trees and the corresponding `fsync` returns `0`, it still could be the case that,
-when reading this data later, it will be found corrupted. Given that the system is already
-replicated for high-availability, it would be wasteful to not use redundancy to repair local storage
-failures, and that's exactly what TigerBeetle does.
+block of an LSM trees and the corresponding `fsync` returns `0` it could still be the case that when 
+reading this data later it will be found to be corrupted. Given that the system is already replicated 
+for high-availability, it would be wasteful to not use redundancy to repair local storage failures. So
+this is exactly what TigerBeetle does.
 
 ## Design Decisions
 
 ### Intentionality in the Design
 
-A lot of software is designed primarily via empirical feedback loop --- apply a small change to the
-software, and, judging the results, revert or double down. For TigerBeetle, we are trying something
-different: think from the first principles what the right solution _should_ be, and then use
+A lot of software is designed primarily via empirical feedback loop. You apply a small change to the
```

Comment:
I like how you remove basically every `---` and just make it two separate sentences! So much better indeed!